app/Controllers/Http/ImageController.js

```diff
@@ -0,0 +1,64 @@
+"use strict";
+const fs = require("fs");
+const Env = use("Env");
+const AWS = require("aws-sdk");
+if (
+  process.env.AWS_ACCESS_KEY_ID &&
+  process.env.AWS_SECRET_ACCESS_KEY &&
+  process.env.AWS_REGION
+) {
+  // do nothing, the aws-sdk will read those environment variables
+} else {
+  AWS.config.credentials = new AWS.SharedIniFileCredentials({
+    profile: Env.get("AWS_PROFILE_NAME", "default"),
+  });
+}
+const awsCloudFrontSigner = require("aws-cloudfront-sign");
+
+class ImageController {
+  // uploads image to S3, returns image url
+  async uploadImage({ request, response }) {
+    const { region } = request.all();
+
+    AWS.config.update({ region });
+
+    const s3 = new AWS.S3();
+
+    try {
+      // get bucket name from environment variables
+      const bucket = Env.get("AWS_S3_BUCKET_NAME");
+
+      const file = request.file("file");
+      // use file temp path to create a buffer
+      const buffer = await fs.promises.readFile(file.tmpPath);
+
+      const key = file.clientName;
+
+      await s3
+        .putObject({
+          Key: key,
+          Bucket: bucket,
+          Body: buffer,
+        })
+        .promise();
+
+      const signingParams = {
```

Comment:
I used the signing approach to get the CDN, which requires a public key and a private key, which allows expire date in case it is needed, I can use the normal Prefix-only way instead.

Reply:
Yes, we will use the prefix only method as per the task requirements.